src/pybind/mgr/dashboard/frontend/src/app/shared/services/auth-guard.service.ts

```diff
@@ -11,6 +11,9 @@ export class AuthGuardService implements CanActivate, CanActivateChild {
 
   canActivate() {
     if (this.authStorageService.isLoggedIn()) {
+      if (this.router.url === '/user-profile/edit' && this.authStorageService.getForceChangePwd()) {
+        return false;
+      }
       return true;
```

Comment:
IMO you could combine those returns.
```suggestion
       return this.router.url !== '/user-profile/edit' || !this.authStorageService.getForceChangePwd();
```